### PR TITLE
Improve modularized imports docs and add config options to example

### PR DESCRIPTION
**Pull request by chibicode**
This PR improves the [Modularize Imports](https://nextjs.org/docs/advanced-features/compiler#modularize-imports) section of Next.js Compiler docs as well as [the `modularize-imports` example](https://github.com/vercel/next.js/tree/canary/examples/modularize-imports). They were originally added in [this PR](https://github.com/vercel/next.js/pull/34969).

- Link to [the `modularize-imports` example](https://github.com/vercel/next.js/tree/canary/examples/modularize-imports) was missing from the docs.
- Clarified some explanations and reorganized the content so it's easier to understand.
- Added docs for `skipDefaultConversion`.
- Added docs for `preventFullImport`.
- Illustrated how to use `kebabCase` and `skipDefaultConversion` on the [the `modularize-imports` example](https://github.com/vercel/next.js/tree/canary/examples/modularize-imports).

## Documentation / Examples

- [x] Make sure the linting passes by running `pnpm build && pnpm lint`
- [x] The "examples guidelines" are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md)

### CODE REVIEW

File: docs/advanced-features/compiler.md [hunk 1]
```diff
@@ -312,13 +319,36 @@ module.exports = {
...
 
-- Using regular expressions
+```js
+// Before
+import { MyModule } from 'my-library'
 
-Similar to `babel-plugin-transform-imports`, but the transform is templated with [handlebars](https://docs.rs/handlebars) and regular expressions are in Rust [regex](https://docs.rs/regex/latest/regex/) crate's syntax.
+// After (`MyModule` was converted to `my-module`)
+import MyModule from 'my-library/my-module'
+```
```
Comment (chibicode) [inline docs/advanced-features/compiler.md:349]: Added an example of how to use the helper functions (e.g. `kebabCase`) — previously, you had to look up the linked handlebars documentation to learn how to use helper functions.

Also, I used the `my-library`, `MyModule`, etc example which was already used in the regex example that follows.

File: docs/advanced-features/compiler.md [hunk 2]
```diff
@@ -312,13 +319,36 @@ module.exports = {
...
 
-The config:
+You can also use regular expressions using Rust [regex](https://docs.rs/regex/latest/regex/) crate’s syntax:
```
Comment (chibicode) [inline docs/advanced-features/compiler.md:351]: This was previously:

> Similar to `babel-plugin-transform-imports`, but the transform is templated with [handlebars](https://docs.rs/handlebars) and regular expressions are in Rust [regex](https://docs.rs/regex/latest/regex/) crate's syntax.

I made it much shorter, as babel-plugin-transform-imports` and handlebars were already mentioned previously.

File: docs/advanced-features/compiler.md [hunk 3]
```diff
@@ -331,30 +361,77 @@ module.exports = {
 }
 ```
 
-Cause this code:
+The above config will transform your code as follows:
```
Comment (chibicode) [inline docs/advanced-features/compiler.md:364]: I made the examples more consistent by following this pattern throughout this page:

> ```js
> // next.config.js
> ...
> ```
>
> The above config will transform your code as follows:
>
> ```js
> // Before
> ...
>
> // After
> ...
> ```

File: docs/going-to-production.md [hunk 4]
```diff
@@ -100,7 +100,7 @@ Each file inside your `pages/` directory will automatically be code split into i
 <details open>
   <summary><b>Examples</b></summary>
   <ul>
-    <li><a href="https://github.com/Logflare/next-pino-logflare-logging-example">with-logging</a></li>
+    <li><a href="https://github.com/Logflare/next-pino-logflare-logging-example">Pino and Logflare Example</a></li>
```
Comment (chibicode) [inline docs/going-to-production.md:103]: This was unrelated, but I found that the example name was wrong, so I fixed it. Link: https://github.com/Logflare/next-pino-logflare-logging-example

File: examples/modularize-imports/next.config.js [hunk 5]
```diff
@@ -8,6 +8,13 @@ const nextConfig = {
     '../components/halves': {
       transform: '../components/halves/{{ member }}',
     },
+    '../components/ui': {
+      // Use `kebabCase` helper to convert `HeadingLarge` to `heading-large`
+      transform: '../components/ui/{{ kebabCase member }}',
+      // Use `skipDefaultConversion`, since `HeadingLarge` is exported
+      // using named exports instead of default exports
+      skipDefaultConversion: true,
+    },
```
Comment (chibicode) [inline examples/modularize-imports/next.config.js:17]: Adds an example of `kebabCase` and `skipDefaultConversion` to the example.

File: docs/advanced-features/compiler.md [hunk 6]
```diff
@@ -312,13 +319,36 @@ module.exports = {
...
 ```
 
-Advanced transformations:
+#### Handlebars variables and helper functions
+
+This transform uses [handlebars](https://docs.rs/handlebars) to template the replacement import path in the `transform` field. These variables and helper functions are available:
+
+1. `member`: Has type `string`. The name of the member import.
+2. `lowerCase`, `upperCase`, `camelCase`, `kebabCase`: Helper functions to convert a string to lower, upper, camel or kebab cases.
+3. `matches`: Has type `string[]`. All groups matched by the regular expression. `matches.[0]` is the full match.
```
Comment (chibicode) [inline docs/advanced-features/compiler.md:328]: I reordered these. Previously it was:

> 1. `matches`: ...
> 2. `member`: ...
> 3. `lowerCase`, `upperCase`, `camelCase`, `kebabCase`: ...

But you'd likely use use `member` the most, then case helpers. So I re-ordered.

File: docs/advanced-features/compiler.md [hunk 7]
```diff
@@ -331,30 +361,77 @@ module.exports = {
...
+
+The above config will transform your code as follows:
+
+```js
+// Before
+import { MyModule } from 'my-library'
+
+// After (imports `MyModule` using named import)
+import { MyModule } from 'my-library/MyModule'
+```
```
Comment (chibicode) [inline docs/advanced-features/compiler.md:413]: Added docs for `skipDefaultConversion` ([Next.js type](https://github.com/vercel/next.js/blob/90b86be6dc50287b589363dc71a89f8bb7600736/packages/next/src/server/config-shared.ts#L507) / [SWC implementation](https://github.com/swc-project/plugins/blob/221ec0d713346c5460cae6c789f29ff9dcb07897/packages/transform-imports/transform/src/lib.rs#L84-L85)) which was undocumenented.

I tested this locally.

File: docs/advanced-features/compiler.md [hunk 8]
```diff
@@ -331,30 +361,77 @@ module.exports = {
...
-```js
+// After
 import MyModule from 'my-library/MyModule'
 import App from 'my-library/components/App'
 import Header from 'my-library/components/App/Header'
 import Footer from 'my-library/components/App/Footer'
 ```
 
-- Handlebars templating
+#### Using named imports
```
Comment (chibicode) [inline docs/advanced-features/compiler.md:379]: Added docs for `skipDefaultConversion` ([Next.js type](https://github.com/vercel/next.js/blob/90b86be6dc50287b589363dc71a89f8bb7600736/packages/next/src/server/config-shared.ts#L507) / [SWC implementation](https://github.com/swc-project/plugins/blob/221ec0d713346c5460cae6c789f29ff9dcb07897/packages/transform-imports/transform/src/lib.rs#L84-L85)) which was not documented.

I tested this locally.

File: docs/advanced-features/compiler.md [hunk 9]
```diff
@@ -331,30 +361,77 @@ module.exports = {
...
+
+```js
+// Before
+import { MyModule } from 'my-library'
+
+// After (imports `MyModule` using named import)
+import { MyModule } from 'my-library/MyModule'
+```
+
+#### Preventing full import
```
Comment (chibicode) [inline docs/advanced-features/compiler.md:415]: Added docs for `preventFullImport` ([Next.js type](https://github.com/vercel/next.js/blob/90b86be6dc50287b589363dc71a89f8bb7600736/packages/next/src/server/config-shared.ts#L506) / [SWC implementation](https://github.com/swc-project/plugins/blob/221ec0d713346c5460cae6c789f29ff9dcb07897/packages/transform-imports/transform/src/lib.rs#L101)) which was not documented.

I tested this locally.

**Review by ijjk — APPROVED** [review]
Example and docs changes look great, thanks!

**Pull request merged**